### ISSUE TITLE
Input Validation: Allowing Alphabetic Characters in Invoice Creation and Payment Fields

### ISSUE THREAD

**Issue by shubhamkmr04** [ZeusLN/zeus#2132]
### Describe your enhancement idea

On the send view, users are able to input alphabetic characters when creating an invoice, which should not be allowed. Similarly, when attempting to pay an invoice with alphabetic characters, users receive an error message stating 'amount must be specified when paying a zero amount invoice'.

**Comment by kaloudis** [comment]
We can limit the input here to be `numeric` - it should be a one line fix

**Comment by kaloudis** [comment]
It appears the keyboards are set appropriately here, it's only with a full text keyboard (like in dev) that you can get around this - not sure it's worth adding extra logic here

**Comment by shubhamkmr04** [comment]
Yeah can't recreate it in my phone, but one issue that might be there is we shouldn't be able to proceed with 0 sats input which results in `bad argument ` error

**Comment by nisargpatel7042lva** [comment]
Hi, I'd like to take this. Two separate fixes needed here: on the invoice creation side, the amount field likely isn't constrained to a numeric keyboard/input filter, so I'll add that plus a regex guard on change. On the payment side, the confusing "amount must be specified when paying a zero amount invoice" message suggests that when alphabetic input fails to parse (NaN), it's falling through to the same code path as an actual zero-amount invoice rather than being caught as an invalid-input case earlier. I'll add an explicit NaN/invalid check before that branch so it surfaces a clear "invalid amount" error instead. Happy to be assigned.
